caddyhttp/proxy/upstream.go

```diff
@@ -434,6 +444,14 @@ func (u *staticUpstream) GetHostCount() int {
 	return len(u.Hosts)
 }
 
+// Stop sends a signal to all running goroutines to exit and waits for them to
+// finish before returning.
```

Comment:
Should be obvious already, but maybe clarify in the comment that this stops all goroutines that were started by this staticUpstream.